src/legacy/core_plugins/console/index.js

```diff
@@ -18,24 +18,54 @@
  */
 
 import Boom from 'boom';
+import { first } from 'rxjs/operators';
 import { resolve, join, sep } from 'path';
 import url from 'url';
-import { has, isEmpty, head } from 'lodash';
+import { has, isEmpty, head, pick, isPlainObject } from 'lodash';
 
 import { resolveApi } from './api_server/server';
 import { addExtensionSpecFilePath } from './api_server/spec';
-import setHeaders from '../elasticsearch/lib/set_headers';
 
 import {
   ProxyConfigCollection,
   getElasticsearchProxyConfig,
   createProxyRoute
 } from './server';
 
+function filterHeaders(originalHeaders, headersToKeep) {
```

Comment:
__note:__ since it's the only place in Kibana that consumed `filterHeaders` and `setHeaders` I just moved it here and remove from ES plugin public interface.

Ideally we shouldn't have this logic in two places though...